internal/v4/stats.go

```diff
@@ -132,17 +132,35 @@ func (h *ReqHandler) serveStatsUpdate(_ http.Header, r *http.Request) (interface
 		return nil, errgo.Notef(err, "cannot unmarshal body")
 	}
 
-	rid, err := h.resolveURL(req.CharmReference)
-	if err != nil {
-		return nil, errgo.Mask(err, errgo.Is(params.ErrNotFound))
-	}
+	errors := make([]error, 0)
+	for i := range req {
+		rid, err := h.resolveURL(req[i].CharmReference)
```

Comment:
Note that I don't think resolveURL does anything if the URL is already full specified, which means that it would be possible to game the system by adding stats for non-existent charms and revisions. Perhaps we should actually try to retrieve the charm?